src/components/TimelineContainer.jsx

```diff
@@ -1,89 +1,43 @@
-import TweetContainer from "./TweetContainer.jsx";
+import TweetContainer from "./TweetContainer";
 
 class TimelineContainer extends Component {
     constructor(props) {
         super(props);
-
-        this.state = {
-            data: [],
-            errorOccurred: false
-        }
-        
-        this.errorMessage = "This content is not currently available. Please try again later.";
-        this.fillerMessage = "Loading your Twitter timeline...";
-
-        this.fetchData = () => {
-            fetchTimeline()
-            .then((data) => {
-                this.updateStatus(data);
-            })
-            .catch((error) => {
-                console.log(`Error occurred during fetchData: ${error}`);
-                this.updateStatus([]);
-            });
-        }
-
-        this.updateStatus = (responseData) => {
-            if(responseData.length === 0) {
-                this.setState({
-                    data: responseData,
-                    errorOccurred: true
-                });
-            } else {
-                this.setState({
-                    data: responseData,
-                    errorOccurred: false
-                });
-            }
-        }
-
-        this.handleClick = (e) => {
-            e.preventDefault();
-            this.setState({
-                data: [],
-                errorOccurred: false
-            });
-            this.fetchData();
-        }
-    }
-
-    componentDidMount() {
-        this.fetchData();
     }
 
     render() {
-        let timelineContainer;
-        if(this.state.errorOccurred === true) {
+        let timelineContainer = <div></div>;
+
+        if(!this.props.fetchComplete) {
             timelineContainer = (
-                <div id="timelineContainer">
-                    <p>{this.errorMessage}</p>
+                <div className="timelineContainer">
+                    <p>{this.loadingMessage}</p>
                 </div>
             );
-        } else if(this.state.errorOccurred === false && this.state.data.length > 0) {
-            timelineContainer = (
-                <div id="timelineContainer">
-                    {this.state.data.map(status => 
-                        <TweetContainer key={status.postUrl} user={status.user} postUrl={status.postUrl} message={status.message} createdAt={status.createdAt}/>
-                    )}
-                </div> 
-            );
-        } else if (this.state.errorOccurred === false) {
+        } else if (this.props.fetchComplete && this.props.errorOccurred) {
             timelineContainer = (
-                <div id="timelineContainer">
-                    <p>{this.fillerMessage}</p>
+                <div className="timelineContainer">
+                    <p>{this.errorMessage}</p>
                 </div>
             );
+        } else if (this.props.fetchComplete && !this.props.errorOccurred) {
+            timelineContainer = (this.props.data.length === 0)?
```

Comment:
any chance data may not exist?

Reply:
I don't believe so. Data is passed from the usertimeline & hometimeline container states. In those states data is initially an empty array. It is then explicitly set only twice: when the button is click (set to empty array) and when the api call is successful.

However, just in case data returns something other than an array, I have added a check for it in Timeline Container.